ee/modules/021-cni-cilium/hooks/ee/types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2024 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package ee
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var hashFunc func(key string) [32]byte
+
+func init() {
+	hashFunc = newHashFunc()
+}
+
+type EgressGatewayState struct {
+	Generation                  int64             `json:"generation"`
+	Name                        string            `json:"name"`
+	Mode                        string            `json:"mode"`
+	IP                          string            `json:"ip"`
+	RoutingTableName            string            `json:"routingTableName"`
+	UID                         types.UID         `json:"uid,omitempty"`
+	DesiredActiveNode           string            `json:"desiredActiveNode"`
+	AllNodes                    []string          `json:"allNodes"`
+	ReadyNodes                  []string          `json:"readyNodes"`
+	HealthyNodes                []string          `json:"healthyNodes"`
+	HealthyNodesWithEgressAgent []string          `json:"healthyNodesWithEgressAgent"`
+	CurrentActiveNodes          []string          `json:"currentActiveNodes"`
+	NodeSelector                map[string]string `json:"nodeSelector"`
+}
+
+func (eg *EgressGatewayState) electDesiredActiveNode() string {
```

Comment:
The method is private but the structure is public. Seems like an error.

Reply:
EgressGatewayState is private, yes.